service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/transport/amqps/AmqpFileUploadNotificationReceivedHandler.java

```diff
@@ -63,6 +76,26 @@
         add(new FlowController());
     }
 
+    AmqpFileUploadNotificationReceivedHandler(
+            String hostName,
+            TokenCredential authenticationTokenProvider,
+            TokenCredentialType authorizationType,
+            IotHubServiceClientProtocol iotHubServiceClientProtocol,
+            AmqpFeedbackReceivedEvent amqpFeedbackReceivedEvent,
+            ProxyOptions proxyOptions,
+            SSLContext sslContext)
+    {
+        super(hostName, authenticationTokenProvider, authorizationType, iotHubServiceClientProtocol, proxyOptions, sslContext);
+
+        this.amqpFeedbackReceivedEvent = amqpFeedbackReceivedEvent;
```

Comment:
Line 90-97 seems to same across constructors and other handlers. Can this be moved to base class constructor?

Reply:
Good idea. I can move the handshaker and flowcontroller to the abstract class rather than doing them in all these constructors